addon-test-support/setup-intl.ts

```diff
@@ -16,24 +27,25 @@ import { missingMessage } from './-private/serialize-translation';
  * @param {string} [locale]
  * @param {object} [translations]
  */
-export default function setupIntl(hooks, locale, translations) {
-  if (typeof locale === 'object' && !Array.isArray(locale)) {
-    translations = locale;
-    locale = null;
-  }
-
-  hooks.beforeEach(function() {
+export default function setupIntl(
+  hooks: Hooks,
+  locale?: string | string[] | Translations,
+  translations?: Translations
+) {
+  hooks.beforeEach(function(this: ApplicationContext) {
     this.owner.register('util:intl/missing-message', missingMessage);
     this.intl = this.owner.lookup('service:intl');
   });
 
-  if (locale) {
-    hooks.beforeEach(function() {
-      this.intl.setLocale(locale);
+  if (typeof locale === 'string' || Array.isArray(locale)) {
+    hooks.beforeEach(function(this: ApplicationContext) {
+      this.intl!.setLocale(locale as string | string[]);
     });
+  } else if (typeof locale === 'object') {
+    hooks.beforeEach(() => addTranslations(locale, undefined));
   }
 
   if (translations) {
-    hooks.beforeEach(() => addTranslations(translations));
+    hooks.beforeEach(() => addTranslations(translations, undefined));
```

Comment:
Are the explicit `undefined` arguments necessary?

I would expect the `translations?` to work correctly. Maybe it works with the suggested `...args` change to `makeIntlHelper`? 🤔 

```ts
export default makeIntlHelper((intl: IntlService, localeName: string | Translations, translations?: Translations) => { ... });
```